File: main/java/org/goafabric/core/data/repository/entity/PractitionerEo.java
package org.goafabric.core.data.repository.entity;

import jakarta.persistence.*;
import org.goafabric.core.data.repository.extensions.AuditListener;

import java.time.LocalDate;

@Entity
@Table(name = "practitioner")
@EntityListeners(AuditListener.class)
public class PractitionerEo {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    public String id;

    public String givenName;

    public String familyName;

    public String gender;

    public LocalDate birthDate;

    @OneToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id", referencedColumnName = "id")
    public AddressEo address;

    @OneToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "contact_point_id", referencedColumnName = "id")
    public ContactPointEo contactPoint;

    @Version //optimistic locking
    public Long version;

}

File: src/main/kotlin/org/goafabric/core/organization/persistence/entity/PractitionerEo.kt
package org.goafabric.core.organization.persistence.entity

import jakarta.persistence.*
import org.goafabric.core.persistence.extensions.AuditTrailListener
import org.goafabric.core.persistence.extensions.KafkaPublisher
import java.time.LocalDate

@Entity
@Table(name = "practitioner")
@EntityListeners(AuditTrailListener::class, KafkaPublisher::class)
class PractitionerEo(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: String? = null,

    var givenName: String? = null,
    var familyName: String? = null,
    var gender: String? = null,
    var birthDate: LocalDate? = null,
    var lanr: String? = null,

    @OneToMany(cascade = [CascadeType.ALL])
    @JoinColumn(name = "practitioner_id")
    var address: MutableList<AddressEo> = mutableListOf(),

    @OneToMany(cascade = [CascadeType.ALL])
    @JoinColumn(name = "practitioner_id")
    var contactPoint: MutableList<ContactPointEo> = mutableListOf(),

    @Version
    var version: Long? = null
)
